### DISCUSSION TITLE
Opening a web page that points to an app hosted on Vercel, downloads a file instead of opening the app

### DISCUSSION THREAD

**Discussion by akoskm** [vercel/next.js#43028] [Help] [answered]
Hi!

Something very weird is going on with my website https://www.beginneritjobs.com.

When I type the address in the browser's search bar or click the link in my social profiles, it initiates a download instead of opening the page.

![image](https://user-images.githubusercontent.com/3111627/202398059-d36c4e08-b837-4577-93dd-ab1fc951bbcf.png)

I have no idea what's causing this.

The file seems to be a special format that, at first sight, contains some information about my site - not the Google Tag manager that I have installed in <head>

```
M1:{"id":"6507","name":"","chunks":["193:193-3af0fc114aa86630","791:app/skills/[slug]/page-06fc5656888cdf8f"],"async":false}
M2:{"id":"3244","name":"","chunks":["193:193-3af0fc114aa86630","185:app/layout-04b89e0984a7f757"],"async":false}
M3:{"id":"4090","name":"","chunks":["193:193-3af0fc114aa86630","791:app/skills/[slug]/page-06fc5656888cdf8f"],"async":false}
M4:{"id":"9875","name":"default","chunks":["272:webpack-055e237ba82c63fb","17:17-976d5b38d52569cd","744:main-app-5dbe79f35bd947e6"],"async":false}
M5:{"id":"2092","name":"default","chunks":["272:webpack-055e237ba82c63fb","17:17-976d5b38d52569cd","744:main-app-5dbe79f35bd947e6"],"async":false}
J0:[[["",{"children":["",{}]},null,null,true],[[],[["$","link","0",{"rel":"stylesheet","href":"/_next/static/css/cf0b799e0c0cc115.css","precedence":"high"}]],["$","html",null,{"children":[["$","head",null,{"children":[[["$","@1",null,{"strategy":"afterInteractive","async":true,"src":"https://www.googletagmanager.com/gtag/js?
```

Any idea what's going on?

**Comment by icyJoseph** [comment]
That looks like server components props being downloaded... that's very strange... A bit scary how it auto-downloads.

How are you hosting, and structuring your site? Middleware? redirects/rewrites?

**Reply by akoskm** [reply]
The app is hosted on Vercel. I'm using the new `appDir` feature.

Using https://beta.nextjs.org/docs/routing/pages-and-layouts for pages/layouts.

I'm also using https://beta.nextjs.org/docs/api-reference/generate-static-params to generate the static web pages.

I can invite you to the repo if you want to take a look.

**Reply by icyJoseph** [reply]
Sure thing! I am not sure I can contribute so much, but I have another app already migrated to `appDir`, hosted on Vercel, and it doesn't do this.

**Reply by akoskm** [reply]
Cool, thanks for looking into this. I followed the docs, and this had never happened to me before.

It's also super strange because it works now. I only accessed the page with the full URL `https://www.beginneritjobs.com/`.

Now it doesn't download the file. Curious to hear how it works on your end.

**Reply by icyJoseph** [reply]
Yeah I cannot reach `https://beginneritjobs.com` anymore I get redirected to `www`

![Screenshot 2022-11-17 at 10 38 01](https://user-images.githubusercontent.com/21013447/202410623-5588f44a-b6d4-417c-9c42-e0f1d3caaae7.png)

> There's a typo on `JavaScrip` 😅 

Looks like it is solved via infra?

```shell
➜ curl https://beginneritjobs.com/
Redirecting to https://www.beginneritjobs.com/ (308)
```

**Reply by akoskm** [reply]
The redirect is actually a Vercel thing. They recommended setting up the www redirect

![image](https://user-images.githubusercontent.com/3111627/202412066-3c8a79f3-5baf-48e4-81cb-57fc8a2a4c77.png)

The text says JavaScript but the input box size is cutting it off. Thanks! 😅

**Reply by icyJoseph** [reply]
Hehe rushed to conclusions there.

I see, but still kinda `sus`/weird that the server was responding with raw data on the landing route, but alright, I guess the infrastructure switch did help.

**Reply by akoskm** [reply]
Oh, there's been a misunderstanding, I guess.

So this will stop working after some time, I'm 100% sure. I didn't change anything in my setup.

*EDIT*: could you check it _now_, I believe it's broken again.

**Reply by icyJoseph** [reply]
Still seems to work fine.

**Reply by akoskm** [reply]
Ha! Got it

https://user-images.githubusercontent.com/3111627/202445000-63f13667-26a8-4608-a350-1669655e8612.mov

**Reply by akoskm** [reply]
Here's how it looks like when I open it from my Twitter profile

https://user-images.githubusercontent.com/3111627/202446765-bb779ffe-bfdf-4146-b591-6833b53c8dad.mov

**Reply by akoskm** [reply]
Appreciate you help @icyJoseph 🙇 looks like this was an issue that has been fixed!

**Comment by akoskm** [comment, marked as answer]
This was fixed in v13.0.4 https://github.com/vercel/next.js/discussions/42834#discussioncomment-4124532.

**Comment by SaraswatAnushka** [comment]
this is currently happening to my site, I don't know what to do
this is the link https://churn-prediction-eda.vercel.app/
and it downloads this file from wsgi import app
please help

**Reply by akoskm** [reply]
Indeed. Which Next.js are you using currently? Have you tried with the version that worked for me?

**Reply by SaraswatAnushka** [reply]
um this is a python project, from my end i have not used next js or any javascript just 1 simple index.html file that takes user input and sends it to the flask app (or python file, i don't know what to call it)

**Reply by akoskm** [reply]
this discussion is for next.js

**Reply by SaraswatAnushka** [reply]
okay sorry

**Reply by akoskm** [reply]
No worries. Since the issue is clearly reproducible, I'd try to find help in some Vercel communities. Looks like they also have Discussions: https://github.com/vercel/vercel/discussions

**Comment by itsmeeSanj** [comment]
I'm facing the same issue, too. When I clicked on my deployed domain, it downloaded the file instead of opening the URL link.
I'm deploying my React + Vite project.

Can anyone please help me with this issue? I'm a noob in varcel deploying.
Thanks

**Comment by rexsllemel** [comment]
I am facing this too my site is PHP
